### ISSUE TITLE
Fix false positives for attribute selectors in indentation

### ISSUE THREAD

**Issue by benbrandt** [stylelint/stylelint#2791]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where stylelint throws an indentation for rules with [attribute] selectors

> Which rule, if any, is this issue related to?

`indentation`

> What CSS is needed to reproduce this issue?
Wants this:
```css
.btn[disabled],
button[disabled] .material-icons {
  opacity: 0.35;
}
```
to become:
```css
.btn[disabled],
  button[disabled] .material-icons {
  opacity: 0.35;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
    "extends": "stylelint-config-standard",
    "rules": {
      "at-rule-no-vendor-prefix": true,
      "media-feature-name-no-vendor-prefix": true,
      "property-no-vendor-prefix": true,
      "selector-no-vendor-prefix": true,
      "value-no-vendor-prefix": true,
      "property-no-unknown": [
        true,
        {
          "ignoreProperties": [
            "composes"
          ]
        }
      ],
      "selector-type-no-unknown": null
    },
    "ignoreFiles": [
      "coverage/**",
      "dist/**"
    ]
  }
```

> Which version of stylelint are you using?

e.g. `8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint \"**/*.css\"`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No indentation on the second line of the selector

> What actually happened (e.g. what warnings or errors you are getting)?

Expected indentation of 2 spaces (indentation)

**Comment by evilebottnawi** [comment]
@benbrandt why your expected 2 space indentation? Each selector should be start with newline and have 0 spaces? Seems it is non standard indentation.

**Comment by benbrandt** [comment]
@evilebottnawi sorry I may have made it unclear. I do want a newline with 0 spaces, but stylelint is telling me I have to do a newline and 2 spaces. But only on selectors that include [disabled] or something similar.

I realize the normal selector is `:disabled` but this was for overriding some library code. Wasn't sure if this bug also applies to other selectors, but figured I'd bring it up just in case, since it only started showing up in v8

**Comment by plantain-00** [comment]
I got this error too(I expect 0 space, but --fix give me 2 spaces)
it works great with the same code on https://stylelint.io/demo/ , the online demo is not v8.0.0?

@benbrandt 
> Expected indentation of 2 spaces (indentation)
you should change your first post here

**Comment by jeddy3** [comment]
@benbrandt Thanks for reporting this. 

I'm unable to reproduce the behaviour with `8.0.0`, though. 

Both linting and autofixing give:

```css
.btn[disabled],
button[disabled] .material-icons {
  opacity: 0.35;
}
```

For

```json
{
  "rules": {
    "indentation": 2
  }
}
```

Are you able to make a reproducible test repo for us?

**Comment by catearcher** [comment]
@jeddy3 I think you tested it with a CSS file while @benbrandt *probably* used a LESS file.

Autofixing a LESS file containing
```less
.btn[disabled],
button[disabled] .material-icons {
  opacity: 0.35;
}
```

gives me 
```less
.btn[disabled],
  button[disabled] .material-icons {
  opacity: 0.35;
}
```

while autofixing a CSS file with the same content (correctly) does not change a thing.

**Comment by jeddy3** [comment]
@catearcher Thanks for the info! Labelling the issue accordingly. This is likely caused by the interplay between stylelint and the [postcss-less](https://github.com/shellscape/postcss-less) parser, and will require some digging by someone who uses Less. Please consider helping out if you can.

**Comment by benbrandt** [comment]
Yes sorry all, thanks for debugging. It is in a less file

**Comment by bupy7** [comment]
When will have fixed it?

**Comment by jeddy3** [comment]
> Could you say me when will have fixed it?

I don't believe any of the core team use Less in their daily work. As such, we mainly rely on external contributors to Less-related fix issues. @bupy7 If you use Less, please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by vhmth** [comment]
+1 to this :-)

**Comment by HauptmannEck** [comment]
Issue has been opened on [Postcss-Less #96](https://github.com/shellscape/postcss-less/issues/96)

**Comment by HauptmannEck** [comment]
Could people who are interested in this fix jump into the Postcss-less issue above and ask to get the release cut?

**Comment by HauptmannEck** [comment]
The new version 1.1.5 for Postcss-less is released, that includes the fix could a contributor bump the version in the package.json to force at least that version? package.lock files make is hard to grab the new version automatically.

**Comment by jeddy3** [comment]
> The new version 1.1.5 for Postcss-less is released, that includes the fix could a contributor bump the version in the package.json to force at least that version?

Thanks for the update. Sending a PR is the quickest way to get this change in.
